core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/UniqueIDReadWriteLDAPUserStoreManager.java

```diff
@@ -2185,13 +2186,20 @@ protected void modifyUserInRole(String userNameDN, String groupRDN, int modifyTy
                 logger.debug("User: " + userNameDN + " was successfully " + "modified in LDAP group: " + groupRDN);
             }
         } catch (NamingException e) {
-            String errorMessage =
-                    "Error occurred while modifying user entry: " + userNameDN + " in LDAP role: " + groupRDN;
-            if (log.isDebugEnabled()) {
-                log.debug(errorMessage, e);
+            if (e instanceof NameAlreadyBoundException) {
+                // If concurrent requests bypass the existing user validation check, it can throw an exception
+                // regarding unique key violation. Since the user addition is successful, the exception is ignored.
+                log.warn(String.format("Similar entry found when adding the user to LDAP role: %s. Hence skipping " +
+                        "the user addition", groupRDN));
+            } else {
+                String errorMessage =
+                        "Error occurred while modifying user entry: " + userNameDN + " in LDAP role: " + groupRDN;
+                if (log.isDebugEnabled()) {
```

Comment:
Why are we not logging as error?

Reply:
Moved the previous logic to the else part. Hence no need to change it here.